sdk/forms/src/main/java/com/klaviyo/forms/InAppForms.kt

```diff
@@ -41,3 +44,28 @@ fun Klaviyo.registerForInAppForms(
     // And initialize the webview client
     Registry.get<WebViewClient>().initializeWebView()
 }
+
+/**
+ * Halts the in-app forms services and observers,
+ * hiding any currently displayed forms and preventing any further forms from being presented.
+ */
+@UiThread
+fun Klaviyo.unregisterInAppForms() = safeApply {
+    Registry.apply {
+        unregister<InAppFormsConfig>()
+        getOrNull<PresentationManager>()?.dismiss()
+        getOrNull<WebViewClient>()?.destroyWebView()
+    }
+}
+
+/**
+ * Resets the in-app forms listeners with the current configuration.
+ */
+@UiThread
+internal fun Klaviyo.reInitializeInAppForms() = safeApply {
+    // If config is missing, initial registration has not occurred so we can't re-initialize
+    Registry.getOrNull<InAppFormsConfig>()?.let { config ->
+        unregisterInAppForms()
+        registerForInAppForms(config)
+    }
+}
```

Comment:
I tried putting this re-initializing code in a couple different places before settling on an internal extension function.

First in the `PresentationManager`, because dismissing made sense there, but destroy and reinit webview didn't seem like "presenter's" responsibility. And `WebViewClient` also seems inappropriate for the inverse reasons.

Then tried doing this in the observers that trigger re-init -- but felt like it should be a single call from those observers, the observer shouldn't each need to "know" to chain all those things together. 

Then I remembered the `unregisterInAppForms` ticket was coming up, the work is kinda overlapping, so I came up with this approach of reinitializing being the equivalent of un registering and re-registering.